File: JavaScript/0013_roman_to_integer/solution.js
function roman_to_integer(string) {
    const romans = {"I" : 1, "V" : 5, "X" : 10, "L" : 50, "C" : 100, "D" : 500, "M" : 1000};
    let result = [];
    for (let letter of string) {
        if (letter in romans) {
            if (result.length == 0) {
                result.push(romans[letter]);
            } else {
                if (result[result.length - 1] >= romans[letter]) {
                    result.push(romans[letter]);
                } else {
                    result.push(romans[letter] - result[result.length - 1]);
                    result[result.length - 2] = 0;
                }
            }
        }
    }
    const total = result.reduce((acc,val) => acc + val, 0);
    return total;
}

const string = "MCMXCIV";
console.log(roman_to_integer(string));

File: Python/0013_roman_to_integer/solution.py
def roman_to_integer(string):
    romans = {"I" : 1, "V" : 5, "X" : 10, "L" : 50, "C" : 100, "D" : 500, "M" : 1000}
    result = []
    for letter in string:
        if letter in romans.keys():
            if len(result) == 0:
                result.append(romans[letter])
            else:
                if result[-1] >= romans[letter]:
                    result.append(romans[letter])
                else:
                    result.append(romans[letter] - result[-1])
                    result[-2] = 0
    return sum(result)
            

string = "MCMXCIV"
print(roman_to_integer(string))